cmd/azbrowse/main.go

```diff
@@ -341,6 +346,106 @@ func main() {
 	}); err != nil {
 		log.Panicln(err)
 	}
+	if err := g.SetKeybinding("", gocui.KeyCtrlU, gocui.ModNone, func(g *gocui.Gui, v *gocui.View) error {
+
+		item := list.CurrentExpandedItem()
+		if item == nil ||
+			item.SwaggerResourceType == nil ||
+			item.SwaggerResourceType.PutEndpoint == nil {
+			return nil
+		}
+
+		tmpFile, err := ioutil.TempFile(os.TempDir(), "azbrowse-")
+		if err != nil {
+			status.Status(fmt.Sprintf("Cannot create temporary file: %s", err), false)
+			return err
+		}
+
+		err = tmpFile.Close()
+		if err != nil {
+			status.Status(fmt.Sprintf("Cannot close temporary file: %s", err), false)
+			return err
+		}
+
+		newFilename := tmpFile.Name() + ".json"
+		err = os.Rename(tmpFile.Name(), newFilename)
+		if err != nil {
+			status.Status(fmt.Sprintf("Cannot rename temporary file: %s", err), false)
+			return err
+		}
+
+		tmpFile, err = os.Create(newFilename)
+		if err != nil {
+			status.Status(fmt.Sprintf("Cannot open renamed temporary file: %s", err), false)
+			return err
+		}
+
+		// Remember to clean up the file afterwards
+		defer os.Remove(newFilename)
+
+		originalJSON := content.GetContent()
+		clipboard.WriteAll(originalJSON)
+
+		tmpFile.WriteString(originalJSON)
+		tmpFile.Close()
+
+		status.Status("Opening JSON in editor...", false)
+		err = openEditor(tmpFile.Name())
+		if err != nil {
+			status.Status(fmt.Sprintf("Cannot open editor (ensure https://code.visualstudio.com is installed): %s", err), false)
+			return nil
+		}
+
+		updatedJSONBytes, err := ioutil.ReadFile(tmpFile.Name())
+		if err != nil {
+			status.Status(fmt.Sprintf("Cannot open edited file: %s", err), false)
+			return nil
+		}
+
+		updatedJSON := string(updatedJSONBytes)
+		if updatedJSON == originalJSON {
+			status.Status("No changes to JSON - no further action.", false)
+			return nil
+		}
+		if updatedJSON == "" {
```

Comment:
[tweak] We could check that the JSON can be parsed with `json.UnMarshal` before submitting.

Reply:
Good idea - I'd been using invalid JSON as a way to test the error handling, but it makes sense to sanity check before sending :-)